numpy/core/tests/test_umath.py

```diff
@@ -185,6 +185,82 @@ def __array_wrap__(self, arr, context):
 
 
 class TestComparisons:
+    @pytest.mark.parametrize('dtype', np.sctypes['uint'] + np.sctypes['int'] +
+                             np.sctypes['float'] + [np.bool_])
+    def test_comparison_functions(self, dtype):
+        # Initialize input arrays
+        if dtype == np.bool_:
+            a = np.random.choice(a=[False, True], size=1000)
+            b = np.random.choice(a=[False, True], size=1000)
+            scalar = True
+        else:
+            a = np.random.randint(low=1, high=10, size=1000).astype(dtype)
+            b = np.random.randint(low=1, high=10, size=1000).astype(dtype)
+            scalar = 5
+        scalar_np = np.dtype(dtype).type(scalar)
+        a_lst = a.tolist()
+        b_lst = b.tolist()
+
+        # (Binary) Comparison (x1=array, x2=array)
+        lt_b = np.less(a, b)
+        le_b = np.less_equal(a, b)
+        gt_b = np.greater(a, b)
+        ge_b = np.greater_equal(a, b)
+        eq_b = np.equal(a, b)
+        ne_b = np.not_equal(a, b)
+        lt_b_lst = [x < y for x, y in zip(a_lst, b_lst)]
+        le_b_lst = [x <= y for x, y in zip(a_lst, b_lst)]
+        gt_b_lst = [x > y for x, y in zip(a_lst, b_lst)]
+        ge_b_lst = [x >= y for x, y in zip(a_lst, b_lst)]
+        eq_b_lst = [x == y for x, y in zip(a_lst, b_lst)]
+        ne_b_lst = [x != y for x, y in zip(a_lst, b_lst)]
```

Comment:
and then...

```suggestion
        # (Binary) Comparison (x1=array, x2=array)
        comp_b = np_comp(a, b)
        comp_b_list = [py_comp(x, y) for x, y in zip(a_lst, b_lst)]
```

(likewise for `_s1` & `_s2`)